salt/states/glusterfs.py

```diff
@@ -48,57 +49,59 @@ def peered(name):
            'comment': '',
            'result': False}
 
+    # Check if the name resolves to localhost
+    if socket.gethostbyname(name) in __salt__['network.ip_addrs']():
+        ret['result'] = True
+        ret['comment'] = 'Peering with localhost is not needed'
+        return ret
+
     peers = __salt__['glusterfs.list_peers']()
 
-    if peers:
-        if name in peers:
-            ret['result'] = True
-            ret['comment'] = 'Host {0} already peered'.format(name)
-            return ret
-        elif __opts__['test']:
-            ret['comment'] = 'Peer {0} will be added.'.format(name)
-            ret['result'] = None
-            return ret
+    if peers and name in peers:
+        ret['result'] = True
+        ret['comment'] = 'Host {0} already peered'.format(name)
+        return ret
 
```

Comment:
Removing the 'test' option would have broken the unit tests. I have also removed it as I have reworked the unit tests to allow actual logic testing.